### PR TITLE
MINOR: Remove Deadcode in NioTransport CORS

### PR DESCRIPTION
**Pull request by original-brownbear**
* Same as #34134 but for nio transport

### CODE REVIEW

File: plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java [hunk 1]
```diff
@@ -167,11 +167,6 @@ private void setPreflightHeaders(final HttpResponse response) {
     private boolean setOrigin(final HttpResponse response) {
         final String origin = request.headers().get(HttpHeaderNames.ORIGIN);
         if (!Strings.isNullOrEmpty(origin)) {
-            if ("null".equals(origin) && config.isNullOriginAllowed()) {
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java]: This seems needed? Or is this handled in a different way?

File: plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java [hunk 2]
```diff
@@ -201,10 +196,6 @@ private boolean validateOrigin() {
             return true;
         }
 
-        if ("null".equals(origin) && config.isNullOriginAllowed()) {
```
Comment (danielmitterdorfer) [inline plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java]: This seems needed? Or is this handled in a different way?

**Review by danielmitterdorfer — COMMENTED** [review]
I left comments around the removal of null origins.

File: plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java (hunk 2 above)
Comment (original-brownbear) [reply plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java]: @danielmitterdorfer actually this is always `false` (there's no code assigning anything to that field in the builder) so this must be handled a different way or we have a bug on our hands ... (looking into that)
Looking at the Netty code it's unused there as well ... I guess I could drop this logic there also?

Comment (danielmitterdorfer) [reply plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java]: It appears unused indeed and I *think* we don't want/need it but maybe @tbrooks8 or @jasontedor can chime in here?

Comment (original-brownbear) [reply plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/cors/NioCorsHandler.java]: I can't find a single usage of `allowNullOrigin()` in git in any revision. I also don't see why we'd want to support `"null"` for the origin really? (plus it seems this was never set to `true` before anyway)

**Comment by tbrooks8** [comment]
I’ll take a look at this in the next few days. 

For context, I’m pretty sure that this class is copied from netty. So that is why there is unused code. 

At some point we need to consolidate this into a single class that lives in server and operates on the elasticsearch request object.

**Comment by original-brownbear** [comment]
@tbrooks8 ping :)

For what it's worth I'm pretty sure this is copied from Netty since this method was never used so we're probably good removing it here and then eventually just drying this class up as you suggest?

**Comment by tbrooks8** [comment]
I'll take a look at this tomorrow.

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Pull request merged**